python/variable.cpp

```diff
@@ -220,6 +238,20 @@ void bind_init_list(py::class_<Variable> &c) {
                   Dims{label[0]}, Shape{scipp::size(val)},
                   Values(val.begin(), val.end()), units::Unit(unit));
             return variable;
+          } else if (scipp_dtype(dtype) == core::dtype<Eigen::Quaterniond>) {
+            auto val = values.cast<std::vector<Eigen::Quaterniond>>();
+            Variable variable;
+            if (variances) {
+              auto var = variances->cast<std::vector<Eigen::Quaterniond>>();
+              variable = makeVariable<Eigen::Quaterniond>(
+                  Dims{label[0]}, Shape{scipp::size(val)},
+                  Values(val.begin(), val.end()),
+                  Variances(var.begin(), var.end()), units::Unit(unit));
+            } else
+              variable = makeVariable<Eigen::Quaterniond>(
+                  Dims{label[0]}, Shape{scipp::size(val)},
+                  Values(val.begin(), val.end()), units::Unit(unit));
+            return variable;
```

Comment:
These 15 lines are duplicated from above, apart from changing `Vector3d` to `Quaterniond`, I suppose they could be extracted to a templated helper function?